beacon-chain/rpc/beacon_server.go

```diff
@@ -97,24 +99,117 @@ func (bs *BeaconServer) LatestAttestation(req *ptypes.Empty, stream pb.BeaconSer
 	}
 }
 
-// Eth1Data fetches the current ETH 1 data which should be used when voting via
-// block proposal.
-// TODO(1463): Implement this.
+// Eth1Data is a mechanism used by block proposers vote on a recent Ethereum 1.0 block hash and an
+// associated deposit root found in the Ethereum 1.0 deposit contract. When consensus is formed,
+// state.latest_eth1_data is updated, and validator deposits up to this root can be processed.
+// The deposit root can be calculated by calling the get_deposit_root() function of
+// the deposit contract using the post-state of the block hash.
 func (bs *BeaconServer) Eth1Data(ctx context.Context, _ *ptypes.Empty) (*pb.Eth1DataResponse, error) {
-	return &pb.Eth1DataResponse{}, nil
+	beaconState, err := bs.beaconDB.State()
+	if err != nil {
+		return nil, fmt.Errorf("could not fetch beacon state: %v", err)
+	}
+	dataVoteObjects := []*pbp2p.Eth1DataVote{}
+	eth1FollowDistance := int64(params.BeaconConfig().Eth1FollowDistance)
+	for _, vote := range beaconState.Eth1DataVotes {
+		eth1Hash := bytesutil.ToBytes32(vote.Eth1Data.BlockHash32)
+		// Verify the block from the vote's block hash exists in the eth1.0 chain and fetch its height.
+		blockExists, blockNumber, err := bs.powChainService.BlockExists(eth1Hash)
```

Comment:
instead of `blockNumber`, we should try to use `blockHeight` so its consistent with the rest